[2025-05-02T01:54:37.857+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-05-02T01:54:37.933+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: medical_dag_v16.dim_group.creating_dim_procedures manual__2025-05-02T01:20:23.347496+00:00 [queued]>
[2025-05-02T01:54:37.991+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: medical_dag_v16.dim_group.creating_dim_procedures manual__2025-05-02T01:20:23.347496+00:00 [queued]>
[2025-05-02T01:54:37.992+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-05-02T01:54:38.022+0000] {taskinstance.py:2890} INFO - Executing <Task(SparkSubmitOperator): dim_group.creating_dim_procedures> on 2025-05-02 01:20:23.347496+00:00
[2025-05-02T01:54:38.032+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'medical_dag_v16', 'dim_group.creating_dim_procedures', 'manual__2025-05-02T01:20:23.347496+00:00', '--job-id', '163', '--raw', '--subdir', 'DAGS_FOLDER/raw_enriched_dag.py', '--cfg-path', '/tmp/tmpvms55_4n']
[2025-05-02T01:54:38.036+0000] {standard_task_runner.py:105} INFO - Job 163: Subtask dim_group.creating_dim_procedures
[2025-05-02T01:54:38.038+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=4880) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-05-02T01:54:38.039+0000] {standard_task_runner.py:72} INFO - Started process 4881 to run task
[2025-05-02T01:54:38.105+0000] {task_command.py:467} INFO - Running <TaskInstance: medical_dag_v16.dim_group.creating_dim_procedures manual__2025-05-02T01:20:23.347496+00:00 [running]> on host airflow-scheduler
[2025-05-02T01:54:38.235+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='medical_dag_v16' AIRFLOW_CTX_TASK_ID='dim_group.creating_dim_procedures' AIRFLOW_CTX_EXECUTION_DATE='2025-05-02T01:20:23.347496+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-02T01:20:23.347496+00:00'
[2025-05-02T01:54:38.236+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-05-02T01:54:38.278+0000] {base.py:84} INFO - Retrieving connection 'spark_conn'
[2025-05-02T01:54:38.280+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_procedures.py
[2025-05-02T01:54:40.804+0000] {spark_submit.py:649} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-05-02T01:54:40.924+0000] {spark_submit.py:649} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-05-02T01:54:40.926+0000] {spark_submit.py:649} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-05-02T01:54:40.934+0000] {spark_submit.py:649} INFO - io.delta#delta-spark_2.12 added as a dependency
[2025-05-02T01:54:40.936+0000] {spark_submit.py:649} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-05-02T01:54:40.937+0000] {spark_submit.py:649} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-a5d7210b-8e36-484c-9c6c-fef195b08d00;1.0
[2025-05-02T01:54:40.937+0000] {spark_submit.py:649} INFO - confs: [default]
[2025-05-02T01:54:41.162+0000] {spark_submit.py:649} INFO - found io.delta#delta-spark_2.12;3.3.0 in central
[2025-05-02T01:54:41.217+0000] {spark_submit.py:649} INFO - found io.delta#delta-storage;3.3.0 in central
[2025-05-02T01:54:41.266+0000] {spark_submit.py:649} INFO - found org.antlr#antlr4-runtime;4.9.3 in central
[2025-05-02T01:54:41.375+0000] {spark_submit.py:649} INFO - found org.apache.hadoop#hadoop-aws;3.3.4 in central
[2025-05-02T01:54:41.406+0000] {spark_submit.py:649} INFO - found com.amazonaws#aws-java-sdk-bundle;1.12.262 in central
[2025-05-02T01:54:41.442+0000] {spark_submit.py:649} INFO - found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-05-02T01:54:41.483+0000] {spark_submit.py:649} INFO - :: resolution report :: resolve 524ms :: artifacts dl 23ms
[2025-05-02T01:54:41.484+0000] {spark_submit.py:649} INFO - :: modules in use:
[2025-05-02T01:54:41.485+0000] {spark_submit.py:649} INFO - com.amazonaws#aws-java-sdk-bundle;1.12.262 from central in [default]
[2025-05-02T01:54:41.486+0000] {spark_submit.py:649} INFO - io.delta#delta-spark_2.12;3.3.0 from central in [default]
[2025-05-02T01:54:41.487+0000] {spark_submit.py:649} INFO - io.delta#delta-storage;3.3.0 from central in [default]
[2025-05-02T01:54:41.488+0000] {spark_submit.py:649} INFO - org.antlr#antlr4-runtime;4.9.3 from central in [default]
[2025-05-02T01:54:41.489+0000] {spark_submit.py:649} INFO - org.apache.hadoop#hadoop-aws;3.3.4 from central in [default]
[2025-05-02T01:54:41.490+0000] {spark_submit.py:649} INFO - org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-05-02T01:54:41.491+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T01:54:41.492+0000] {spark_submit.py:649} INFO - |                  |            modules            ||   artifacts   |
[2025-05-02T01:54:41.493+0000] {spark_submit.py:649} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-05-02T01:54:41.493+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T01:54:41.494+0000] {spark_submit.py:649} INFO - |      default     |   6   |   0   |   0   |   0   ||   6   |   0   |
[2025-05-02T01:54:41.498+0000] {spark_submit.py:649} INFO - ---------------------------------------------------------------------
[2025-05-02T01:54:41.499+0000] {spark_submit.py:649} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-a5d7210b-8e36-484c-9c6c-fef195b08d00
[2025-05-02T01:54:41.500+0000] {spark_submit.py:649} INFO - confs: [default]
[2025-05-02T01:54:41.513+0000] {spark_submit.py:649} INFO - 0 artifacts copied, 6 already retrieved (0kB/13ms)
[2025-05-02T01:54:41.969+0000] {spark_submit.py:649} INFO - 25/05/02 01:54:41 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-02T01:54:45.084+0000] {spark_submit.py:649} INFO - Traceback (most recent call last):
[2025-05-02T01:54:45.086+0000] {spark_submit.py:649} INFO - File "/opt/etl/enriched_curated/creating_dim_procedures.py", line 200, in <module>
[2025-05-02T01:54:45.094+0000] {spark_submit.py:649} INFO - main()
[2025-05-02T01:54:45.095+0000] {spark_submit.py:649} INFO - File "/opt/etl/enriched_curated/creating_dim_procedures.py", line 8, in main
[2025-05-02T01:54:45.100+0000] {spark_submit.py:649} INFO - with open('../config.yaml') as config_file:
[2025-05-02T01:54:45.102+0000] {spark_submit.py:649} INFO - ^^^^^^^^^^^^^^^^^^^^^^
[2025-05-02T01:54:45.103+0000] {spark_submit.py:649} INFO - FileNotFoundError: [Errno 2] No such file or directory: '../config.yaml'
[2025-05-02T01:54:45.213+0000] {spark_submit.py:649} INFO - 25/05/02 01:54:45 INFO ShutdownHookManager: Shutdown hook called
[2025-05-02T01:54:45.214+0000] {spark_submit.py:649} INFO - 25/05/02 01:54:45 INFO ShutdownHookManager: Deleting directory /tmp/spark-9d7361f7-1f02-4f0c-9ebc-5d704ba2cc97
[2025-05-02T01:54:45.276+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 575, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_procedures.py. Error code is: 1.
[2025-05-02T01:54:45.285+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=medical_dag_v16, task_id=dim_group.creating_dim_procedures, run_id=manual__2025-05-02T01:20:23.347496+00:00, execution_date=20250502T012023, start_date=20250502T015437, end_date=20250502T015445
[2025-05-02T01:54:45.335+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-05-02T01:54:45.336+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 163 for task dim_group.creating_dim_procedures (Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_procedures.py. Error code is: 1.; 4881)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 575, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --packages io.delta:delta-spark_2.12:3.3.0,org.apache.hadoop:hadoop-aws:3.3.4 --name arrow-spark --deploy-mode client /opt/etl/enriched_curated/creating_dim_procedures.py. Error code is: 1.
[2025-05-02T01:54:45.381+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-05-02T01:54:45.409+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-02T01:54:45.412+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
